lib/aruba/extensions/object/deep_dup.rb

```diff
@@ -0,0 +1,46 @@
+require 'aruba/extensions/object/duplicable'
+
+class Object
+  # Returns a deep copy of object if it's duplicable. If it's
+  # not duplicable, returns +self+.
+  #
+  #   object = Object.new
+  #   dup    = object.deep_dup
+  #   dup.instance_variable_set(:@a, 1)
+  #
+  #   object.instance_variable_defined?(:@a) # => false
+  #   dup.instance_variable_defined?(:@a)    # => true
+  def deep_dup
```

Comment:
Can we just use

`Marshal.load(Marshal.dump(obj))`

instead of monkey patching.